### PR TITLE
Update wasm image

### PR DESCRIPTION
**Pull request by Meakk**
- Update docker actions to latest versions
- Split wasm docker image in multi-stages
- Add a stage to build Assimp
- Add `ARG` variables that one can set with `--build-arg arg=val` or using the `build-args` key in the action

### CODE REVIEW

File: .github/workflows/build_docker_image.yml [hunk 1]
```diff
@@ -33,32 +33,35 @@ jobs:
...
     - name: Build and push
-      uses: docker/build-push-action@v3
+      uses: docker/build-push-action@v6
       with:
         push: ${{ github.ref == 'refs/heads/main' }}
         tags: ghcr.io/f3d-app/f3d-wasm:latest
         file: webassembly/Dockerfile
+        build-args: |
+          CMAKE_BUILD_PARALLEL_LEVEL=4
+          VTK_VERSION=1fb0e032df606c64c37448edccf24825e45fb8a4
```
Comment (mwestphal) [inline .github/workflows/build_docker_image.yml:50]: ASSIMP_VERSION=v5.4.2 ?

File: .github/workflows/build_docker_image.yml [hunk 2]
```diff
@@ -33,32 +33,35 @@ jobs:
...
 
     - name: Build and push
-      uses: docker/build-push-action@v3
+      uses: docker/build-push-action@v6
       with:
         push: ${{ github.ref == 'refs/heads/main' }}
         tags: ghcr.io/f3d-app/f3d-wasm:latest
         file: webassembly/Dockerfile
+        build-args: |
+          CMAKE_BUILD_PARALLEL_LEVEL=4
```
Comment (mwestphal) [inline .github/workflows/build_docker_image.yml:49]: specific reason to bring it down to 4 ?

**Review by mwestphal — COMMENTED** [review]
some questions

File: .github/workflows/build_docker_image.yml (hunk 2 above)
Comment (Meakk) [reply .github/workflows/build_docker_image.yml:49]: Linux runners on github have 4 CPUs now.

Comment (mwestphal) [reply .github/workflows/build_docker_image.yml:49]: ha nice, maybe we should change that in our CI to improve build time

**Review by mwestphal — APPROVED** [review]

**Pull request merged**